### PR TITLE
Provision to add lables to ServiceMonitor resources

### LINKED ISSUE #2535: Add ServiceMonitor Labels customisation so they work out of the box with Kube-Prometheus-Stack
**Is your feature request related to a problem? Please describe.**
The ServiceMonitor templates in the fission-all chart do not currently have any way to customise labels. This would be a good thing to add so monitoring works out of the box for most kube-prometheus-stack configs with a values.yaml change.

**Describe the solution you'd like**
Allow labels to be passed the service monitor. This can be done by either setting the `release` label and populating it from values.yaml or by adding `additionalServiceMonitorLabels` and passing all the values under it.

**Describe alternatives you've considered**
Alternatives are to either remove the LabelSelector from the Prometheus CR, but thats not ideal as there might be multiple Prometheus instances and specific ServiceMonitors might want to be picked. Or to manually edit the generated templates to add in the kube-prometheus-stack release name. Neither are ideal solutions.

### PR DESCRIPTION
**Pull request by runitmisra**
## Description
Added provision to have user-defined lables in the ServiceMonitor resource so monitoring works out of the box for most kube-prometheus-stack configs.

## Which issue(s) this PR fixes:
Fixes #2535 

## Testing
Tested by installing helm chart with these changes and checking if servicemonitors are discovered automatically in prometheus config

## Checklist:

- [x] I ran tests as well as code linting locally to verify my changes. 
- [x] I have done manual verification of my changes, changes working as expected.
- [ ] I have added new tests to cover my changes.
- [x] My changes follow contributing guidelines of Fission.
- [x] I have signed all of my commits.

### CODE REVIEW

File: charts/fission-all/values.yaml [hunk 1]
```diff
@@ -457,6 +457,7 @@ serviceMonitor:
   ##namespace in which you want to deploy servicemonitor
   ##
   namespace: ""
+  additionalServiceMonitorLabels: {}
```
Comment (sanketsudake) [inline charts/fission-all/values.yaml:460]: Can we add a service monitor for all these components ?

I see as of now we have covered 4 out of 6.

```
charts/fission-all/templates/storagesvc/deployment.yaml
21:        prometheus.io/scrape: "true"

charts/fission-all/templates/mqt-fission-kafka/deployment.yaml
22:        prometheus.io/scrape: "true"

charts/fission-all/templates/buildermgr/deployment.yaml
18:        prometheus.io/scrape: "true"

charts/fission-all/templates/controller/deployment.yaml
21:        prometheus.io/scrape: "true"

charts/fission-all/templates/executor/deployment.yaml
18:        prometheus.io/scrape: "true"

charts/fission-all/templates/router/deployment.yaml
27:        prometheus.io/scrape: "true"
```

Comment (sanketsudake) [inline charts/fission-all/values.yaml:460]: Please add description for `additionalServiceMonitorLabels` 
- Whats the purpose?
- How to use it?

Refer rest of values file for reference.

**Review by sanketsudake — CHANGES_REQUESTED** [review]

File: charts/fission-all/values.yaml (hunk 1 above)
Comment (runitmisra) [reply charts/fission-all/values.yaml:462]: Added servicemontiors for the remaining components

Comment (runitmisra) [reply charts/fission-all/values.yaml:462]: Added comment and usage example for the value

**Review by sanketsudake — APPROVED** [review]

**Review by kanuahs — APPROVED** [review]

File: charts/fission-all/values.yaml [hunk 2]
```diff
@@ -458,10 +458,22 @@ serviceMonitor:
...
   ## Map of additional lables to add to the ServiceMonitor resources
-  #  to make prometheus seamlessly discover the ServiceMonitors
+  #  to allow selecting sepcific ServiceMonitors
+  #  in case of multiple prometheus deployments
   additionalServiceMonitorLabels: {}
   #  release: "monitoring"
   #  key: "value"
+podMonitor:
+  enabled: false
+  ##namespace in which you want to deploy servicemonitor
```
Comment (sanketsudake) [inline charts/fission-all/values.yaml:468]: podmonitor ?

**Review by sanketsudake — CHANGES_REQUESTED** [review]

**Comment by sanketsudake** [comment]
Closing PR as changes merged via https://github.com/fission/fission/pull/2541

**Pull request closed without merging**